components/org.wso2.carbon.micro.integrator.extensions/org.wso2.carbon.micro.integrator.management.apis/src/main/java/org/wso2/carbon/micro/integrator/management/apis/DataServiceResource.java

```diff
@@ -0,0 +1,158 @@
+
+        // initiate list model
+        DataServicesList dataServicesList = new DataServicesList(dataServicesNames.length);
+
+        for (String dataServiceName : dataServicesNames) {
+            DataService dataService = getDataServiceByName(msgCtx, dataServiceName);
+            ServiceMetaData serviceMetaData = getServiceMetaData(dataService);
+
+            // initiate summary model
+            DataServiceSummary summary = null;
+            if (serviceMetaData != null) {
+                summary = new DataServiceSummary(serviceMetaData.getName(), serviceMetaData.getWsdlURLs());
+            }
+            dataServicesList.addServiceSummary(summary);
+        }
+
+        org.apache.axis2.context.MessageContext axis2MessageContext =
+                ((Axis2MessageContext) msgCtx).getAxis2MessageContext();
+
+
+        String stringPayload = new Gson().toJson(dataServicesList);
+        Utils.setJsonPayLoad(axis2MessageContext, new JSONObject(stringPayload));
+    }
+
+    private void populateDataServiceByName(MessageContext msgCtx, String serviceName) throws AxisFault {
+        DataService dataService = getDataServiceByName(msgCtx, serviceName);
+
+        org.apache.axis2.context.MessageContext axis2MessageContext =
+                ((Axis2MessageContext) msgCtx).getAxis2MessageContext();
+
+        if (dataService == null) {
+            axis2MessageContext.setProperty(Constants.HTTP_STATUS_CODE, Constants.NOT_FOUND);
+        } else {
+            ServiceMetaData serviceMetaData = getServiceMetaData(dataService);
+            DataServiceInfo dataServiceInfo = null;
+
+            if (serviceMetaData != null) {
+                dataServiceInfo = new DataServiceInfo(serviceMetaData.getName(), serviceMetaData.getDescription(),
+                        serviceMetaData.getServiceGroupName(), serviceMetaData.getWsdlURLs());
+
+                Map<String, Query> queries = dataService.getQueries();
+                for (Map.Entry<String, Query> stringQuery : queries.entrySet()) {
+                    QuerySummary querySummary = new QuerySummary(stringQuery.getKey(),
+                            stringQuery.getValue().getNamespace());
+                    dataServiceInfo.addQuery(querySummary);
+                }
+            }
+            String stringPayload = new Gson().toJson(dataServiceInfo);
+            Utils.setJsonPayLoad(axis2MessageContext, new JSONObject(stringPayload));
+        }
+    }
+
+    private DataService getDataServiceByName(MessageContext msgCtx, String serviceName) {
+        AxisService axisService = msgCtx.getConfiguration().
+                getAxisConfiguration().getServiceForActivation(serviceName);
+        DataService dataService = null;
+        if (axisService != null) {
+            dataService = (DataService) axisService.getParameter(DBConstants.DATA_SERVICE_OBJECT).getValue();
+        } else {
+            log.error(String.format("DataService %s is null.", serviceName));
```

Comment:
IMO we do not need to use error level here since this is a user error. We can make it debug.

Reply:
Also you do not have to use `String.format`. You can use the following sysntax.

```java
logger.debug("Value {} was inserted between {} and {}.", newVal, below, above);
```